Tests/FileSystemStorage.cs

```diff
@@ -413,7 +413,12 @@ public void UnlockFile(IntPtr lockFile)
 
         public string GetParentPath(string path)
         {
-            return m_FileSystem.Path.GetDirectoryName(path);
+            string parentPath = m_FileSystem.Path.GetDirectoryName(path);
+            if (parentPath.Length == 0)
+            {
+                return null;
```

Comment:
> Directory information for path, or null if path denotes a root directory or is null. Returns Empty if path does not contain directory information.
https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getdirectoryname?view=net-6.0
```suggestion
                return string.Empty;
```